From 628b4a93c6863b9982d817db6acaacbc4e116453 Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Wed, 21 Nov 2018 20:07:14 +0100
Subject: [PATCH] tests: Call test_cleanup in backtrace-subr.sh
 check_unsupported.

We want to make sure all (temporary) test files are cleaned up even when
we exit 77 to skip the testcase.

https://sourceware.org/bugzilla/show_bug.cgi?id=23901

Tested-by: Kurt Roeckx <kurt@roeckx.be>
Signed-off-by: Mark Wielaard <mark@klomp.org>
---
 tests/ChangeLog         | 5 +++++
 tests/backtrace-subr.sh | 1 +
 2 files changed, 6 insertions(+)

diff --git a/tests/backtrace-subr.sh b/tests/backtrace-subr.sh
index ff42c6ff..53c719df 100644
--- a/tests/backtrace-subr.sh
+++ b/tests/backtrace-subr.sh
@@ -85,6 +85,7 @@ check_unsupported()
   testname=$2
   if grep -q ': Unwinding not supported for this architecture$' $err; then
     echo >&2 $testname: arch not supported
+    test_cleanup
     exit 77
   fi
 }
-- 
2.20.1

